arkouda/io.py

```diff
@@ -831,30 +760,560 @@ def export(
+    Examples
+    --------
+    Read with file Extension
+    >>> x = ak.read('path/name_prefix.h5') # load HDF5 - processing determines file type not extension
+    Read without file Extension
+    >>> x = ak.read('path/name_prefix.parquet', file_format='Parquet') # load Parquet
+    Read Glob Expression
+    >>> x = ak.read('path/name_prefix*') # Reads HDF5
+    """
+    warn(
+        "ak.pdarrayIO.read has been deprecated. Please use ak.IO.read_parquet or ak.IO.read_hdf",
+        DeprecationWarning,
+    )
+    if isinstance(filenames, str):
+        filenames = [filenames]
+    datasets = _prep_datasets(filenames, datasets, allow_errors)
+
+    file_format = file_format.lower()
+    if file_format == "infer":
+        cmd = "readany"
+    elif file_format == "hdf5":
+        cmd = "readAllHdf"
+    elif file_format == "parquet":
+        cmd = "readAllParquet"
+    else:
+        warnings.warn(f"Unrecognized file format string: {file_format}. Inferring file type")
+        cmd = "readany"
+    if iterative:  # iterative calls to server readhdf
+        return {
+            dset: read(
+                filenames,
+                dset,
+                strictTypes=strictTypes,
+                allow_errors=allow_errors,
+                iterative=False,
+                calc_string_offsets=calc_string_offsets,
+            )[dset]
+            for dset in datasets
+        }
+    else:
+        rep_msg = generic_msg(
+            cmd=cmd,
+            args={
+                "strict_types": strictTypes,
+                "dset_size": len(datasets),
+                "filename_size": len(filenames),
+                "allow_errors": allow_errors,
+                "calc_string_offsets": calc_string_offsets,
+                "dsets": datasets,
+                "filenames": filenames,
+            },
         )
+        rep = json.loads(rep_msg)  # See GenSymIO._buildReadAllHdfMsgJson for json structure
+        items = rep["items"] if "items" in rep else []
+        file_errors = rep["file_errors"] if "file_errors" in rep else []
+        if allow_errors and file_errors:
+            file_error_count = rep["file_error_count"] if "file_error_count" in rep else -1
+            warnings.warn(
+                f"There were {file_error_count} errors reading files on the server. "
+                + f"Sample error messages {file_errors}",
+                RuntimeWarning,
+            )
 
-    akdf = DataFrame.load(read_path, file_format=filetype)
-    df = akdf.to_pandas(retain_index=index)
+        # We have a couple possible return conditions
+        # 1. We have multiple items returned i.e. multi pdarrays, multi strings, multi pdarrays & strings
+        # 2. We have a single pdarray
+        # 3. We have a single strings object
+        if len(items) > 1:  # DataSets condition
+            d: Dict[str, Union[pdarray, Strings, arkouda.array_view.ArrayView]] = {}
+            for item in items:
```

Comment:
i think we can use your `_build_objects` helper function here (and for the next ~50 lines of code but I didn't want to highlight it all)

Reply:
Again, I did not update the deprecated functions, but let me know if you think we should.